testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -618,7 +619,13 @@ impl BlockMinerThread {
                     "Miner did not find any transactions to mine, sleeping for {:?}",
                     self.config.miner.empty_mempool_sleep_time
                 );
-                self.reset_nonce_cache = false;
+                // For NextNonceWithHighestFeeRate strategy, keep reset_nonce_cache = true
+                // to ensure considered_txs table is cleared for each block attempt
+                if self.config.miner.mempool_walk_strategy
+                    != MemPoolWalkStrategy::NextNonceWithHighestFeeRate
+                {
+                    self.reset_nonce_cache = false;
+                }
```

Comment:
Can you explain the need for this? I think this indicates a bug somewhere. It's important for performance to avoid clearing the nonce cache when possible.